File: 0413-arithmetic-slices/0413-arithmetic-slices.cpp
class Solution {
public:
    int numberOfArithmeticSlices(vector<int>& nums) {
        
        int n = nums.size();
        if (n<3) return 0;
        vector<int> dp(n,0);
        for (int i=2;i<n;i++){
            if (nums[i]-nums[i-1] == nums[i-1]-nums[i-2]){
                dp[i] = dp[i-1]+1;
            }
        }
        return accumulate(dp.begin(),dp.end(),0);
    }
};

File: 0413-arithmetic-slices/0413-arithmetic-slices.py
class Solution:
    def numberOfArithmeticSlices(self, nums: List[int]) -> int:
        
        dp = [0]*len(nums)
        summ = 0 
        
        for i in range(2,len(nums)):
            if nums[i]-nums[i-1] == nums[i-1] - nums[i-2]:                
                dp[i] = 1+dp[i-1]
                summ+=dp[i]
        return summ
                       
        
        